### PR TITLE
Cleanup two divide-by-zero issues from Coverity

### PR DESCRIPTION
**Pull request by jwrober**
No related issue.

CID 1435482, 1435595

### CODE REVIEW

File: ai/default/aiunit.cpp [hunk 1]
```diff
@@ -641,7 +641,8 @@ static void dai_military_bodyguard(struct ai_type *ait,
       int unit_mv_rate = unit_move_rate(punit);
       fc_assert_ret_msg(unit_mv_rate, "div by zero");
       if (me2goal < me2them
-          || (me2goal / unit_mv_rate < them2goal / unit_move_rate(aunit)
+          || (me2goal / unit_mv_rate
+                  < them2goal / std::min(1, unit_move_rate(aunit))
               && me2goal / unit_mv_rate < me2them / unit_mv_rate
               && unit_mv_rate > unit_move_rate(aunit))) {
```
Comment (lmoureaux) [inline ai/default/aiunit.cpp:647]: We have two techniques to protect against zero `unit_move_rate` now... Can we keep using the same?
```cpp
      int punit_mv_rate = unit_move_rate(punit);
      int aunit_mv_rate = unit_move_rate(aunit);
      fc_assert_ret_msg(punit_mv_rate, "div by zero");
      fc_assert_ret_msg(aunit_mv_rate, "div by zero");
      if (me2goal < me2them
          || (me2goal / punit_mv_rate < them2goal / aunit_move_rate
              && me2goal / unit_mv_rate < me2them / punit_mv_rate
              && unit_mv_rate > aunit_move_rate)) {
```

File: ai/default/aiunit.cpp [hunk 2]
```diff
@@ -640,6 +640,8 @@ static void dai_military_bodyguard(struct ai_type *ait,
       int them2goal = real_map_distance(unit_tile(aunit), aunit->goto_tile);
       int unit_mv_rate = unit_move_rate(punit);
       fc_assert_ret_msg(unit_mv_rate, "div by zero");
+      fc_assert_ret_msg(punit, "div by zero");
+      fc_assert_ret_msg(aunit, "div by zero");
```
Comment (lmoureaux) [inline ai/default/aiunit.cpp:644]: This is not sufficient, even existing units can have a move rate of 0 if `min_speed` is 0 in `units.ruleset`. Consider the following addition to the classic ruleset after setting `min_speed = 0` for Land units:
```ini
; Units get stuck in forests
[effect_elves_catch_units]
type    = "Move_Bonus"
value	= -1
reqs	=
    { "type", "name", "range"
      "Terrain", "Forest", "Local"
    }
```
Now try to play this from the beginning... (and notice the many bugs it exposes)

=> you need to check the move rate as is already done for `punit`

**Comment by lmoureaux** [comment]
Pushed a commit using the same technique for `aunit` and `punit`. This makes the div by zero truly impossible by checking beforehand.

We could also have done some math and used `a/b < c/d  <=>  a*d < b*c` (for positive `b` and `d`), but this has different rounding which might be important.

**Comment by jwrober** [comment]
Thanks for the help on this one.

**Pull request merged**